File: e3/CalculatorTest.java
package com.example;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

public class CalculatorTest {

    private final Calculator calculator = new Calculator();

    @Test
    void add_shouldReturnSum() {
        int result = calculator.add(2, 3);
        assertEquals(5, result);
    }

    @Test
    void subtract_shouldReturnDifference() {
        int result = calculator.subtract(10, 4);
        assertEquals(6, result);
    }

    @Test
    void multiply_shouldReturnProduct() {
        int result = calculator.multiply(3, 5);
        assertEquals(15, result);
    }

    @Test
    void divide_shouldReturnQuotient() {
        int result = calculator.divide(10, 2);
        assertEquals(5, result);
    }

    @Test
    void divide_shouldThrowExceptionWhenDivisorIsZero() {
        assertThrows(IllegalArgumentException.class, () -> calculator.divide(10, 0));
    }
}

File: e2/kotlin/test/kotlin/CalculatorTest.kt
package com.example

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class CalculatorTest {

    private val calculator = Calculator()

    @Test
    fun `add should return sum of two integers`() {
        val result = calculator.add(2, 3)
        assertEquals(5, result)
    }

    @Test
    fun `subtract should return difference of two integers`() {
        val result = calculator.subtract(10, 4)
        assertEquals(6, result)
    }

    @Test
    fun `multiply should return product of two integers`() {
        val result = calculator.multiply(3, 5)
        assertEquals(15, result)
    }

    @Test
    fun `divide should return integer division of two integers`() {
        val result = calculator.divide(10, 2)
        assertEquals(5, result)
    }

    @Test
    fun `divide should throw exception when divisor is zero`() {
        assertThrows(IllegalArgumentException::class.java) {
            calculator.divide(10, 0)
        }
    }
}

